pxemgr/pxemanager_test.go

```diff
@@ -0,0 +1,156 @@
+package pxemgr
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/giantswarm/mayu-infopusher/machinedata"
+	"github.com/giantswarm/mayu/hostmgr"
+)
+
+const (
+	baseConfig = `default_coreos_version: myversion
+network:
+  ip_range:
+    start: 1.1.1.1
+    end: 1.1.1.2
+templates_env:
+  mayu_https_endpoint: https://mayu
+`
+	configOK             = baseConfig + "  storage: mystorage"
+	configErr            = baseConfig
+	lastStageCloudconfig = `before_key: ok
+{{if eq .TemplatesEnv.storage "mystorage"}}storage: mystorage{{end}}
+after_key: ok
+`
+)
+
+var (
+	dir      string
+	fakeEtcd *httptest.Server
+	pxeCfg   PXEManagerConfiguration
+	req      *http.Request
+	w        *httptest.ResponseRecorder
+	cluster  *hostmgr.Cluster
+)
```

Comment:
Very good point, i've added a helper type that wraps all this vars and is test-scoped, pls take a look.